### PR TITLE
Implement MetricsCollector with Timer, Counter and Recorder

### PR DESCRIPTION
**Pull request by trungnq97**

### CODE REVIEW

File: src/main/java/com/haymetric/metrics/MetricsCollector.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    }
+
+    @Override
+    public void stopTimer(final String timerName) {
+        this.metrics.computeIfAbsent(timerName, key -> new Timer()).stopTimer();
+    }
+
+    @Override
+    public void record(final String metricName, final double value, final Unit unit) {
+        this.metrics.computeIfAbsent(metricName, key -> new Recorder(unit)).record(value, unit);
```
Comment (linhdang) [inline src/main/java/com/haymetric/metrics/MetricsCollector.java]: Java 8 or Java 7?

File: src/main/java/com/haymetric/metrics/Recorder.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+            throw new IllegalArgumentException("Unit is not consistent. " + this.unit + " is in use");
+        }
+        values.add(value);
+    }
+
+    @Override
+    public String flushToString() {
+        Double value;
+        final List<Double> values = new ArrayList<>();
+        while ((value = this.values.poll()) != null) {
```
Comment (linhdang) [inline src/main/java/com/haymetric/metrics/Recorder.java]: is this necessary?

File: src/main/java/com/haymetric/metrics/MetricsCollector.java (hunk 1 above)
Comment (trungnq97) [inline src/main/java/com/haymetric/metrics/MetricsCollector.java]: Java 8 adoption is doing quite well. http://www.baeldung.com/java-8-adoption-march-2016
Majority is using Java 8. So let support Java 8 from now on ;)

File: src/main/java/com/haymetric/metrics/Recorder.java (hunk 2 above)
Comment (trungnq97) [inline src/main/java/com/haymetric/metrics/Recorder.java]: Yes, we want to get all values from Recorder to flush.

**Pull request merged**